code/datums/jobs/job/xenomorph.dm

```diff
@@ -38,6 +38,11 @@
 	var/datum/hive_status/normal/HS = GLOB.hive_datums[XENO_HIVE_NORMAL]
 	HS.give_larva_to_next_in_queue()
 
+/datum/job/xenomorph/after_spawn(mob/living/L, mob/M, latejoin)
+	. = ..()
+	var/mob/living/carbon/xenomorph/xeno = L
+	SSminimaps.add_marker(xeno, xeno.z, hud_flags = MINIMAP_FLAG_XENO, iconstate = xeno.xeno_caste.minimap_icon)
+
```

Comment:
Does this not apply the market twice for late joins

Reply:
no